src/js/trove/table.js

```diff
@@ -227,17 +308,52 @@
 
       }
 
+      function makeRowFromValues(vals) {
+        if(headers.length !== vals.length) {
+          throw runtime.ffi.throwRowLengthMismatch(makeTable(headers, []), vals);
+        }
+        return makeRow({ headerIndex: headerIndex }, vals);
+      }
+
       return applyBrand(brandTable, runtime.makeObject({
 
         '_header-raw-array': headers,
         '_rows-raw-array': rows,
 
-        'order-increasing': runtime.makeMethod1(function(_, colname) {
+        'increasing-by': runtime.makeMethod1(function(_, colname) {
+          ffi.checkArity(2, arguments, "increasing-by", true);
+          runtime.checkString(colname);
           return order(true, colname);
         }),
-        'order-decreasing': runtime.makeMethod1(function(_, colname) {
+        'decreasing-by': runtime.makeMethod1(function(_, colname) {
+          ffi.checkArity(2, arguments, "decreasing-by", true);
+          runtime.checkString(colname);
           return order(false, colname);
         }),
+        'order-by': runtime.makeMethod2(function(_, colname, increasing) {
+          ffi.checkArity(3, arguments, "order-by", true);
+          runtime.checkString(colname);
+          runtime.checkBoolean(increasing);
+          return order(increasing, colname);
+        }),
+        'order-by-columns': runtime.makeMethod1(function(_, specs) {
```

Comment:
How is this method different from `multi-order`?  Multi-order takes an array of 2-element arrays, and this takes a list of 2-tuples?  If so, I'd rather
* remove `multi-order` as a Pyret-visible method on tables, in favor of `order-by-columns`,
* keep the implementation of `multiOrder` available as a JS-callable function on tables, and
* change the use of `multi-order` in table-desugaring to call that helper function directly (to keep the desugaring efficient)